### ISSUE TITLE
HubNetworking: Deploy Azure Bastion into Availability Zones

### ISSUE THREAD

**Issue by integyjc** [Azure/ALZ-Bicep#922]
### Describe the feature end to end, including deployment scenario details under which the feature would occur.

The current deployment of bastion doesn't configure zone redundancy. Enabling this feature for regions which support it would increase compliance with ALZ policies around zone resilience and provide a more robust deployment

### Why is this feature important. Describe why this would be important for your organization and others. Would this impact similar orgs in the same way?

Availability, compliance, best practice

### Please provide the correlation id associated with your error or bug.

xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx

### Feature Implementation

```Shell

```

### Check previous GitHub issues

- [x] I have searched the issues for this item and found no duplicate

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by oZakari** [comment]
Hey @sebassem, as availability zones support for Bastion is in public preview as of May. Is this something we want to add to the backlog to add it in the future? Right now, it is pretty limited in what [regions are supported](https://learn.microsoft.com/en-us/azure/reliability/reliability-bastion?toc=%2Fazure%2Fbastion%2FTOC.json#regions-supported) as a heads up.

**Comment by integyjc** [comment]
> Hey [@sebassem](https://github.com/sebassem), as availability zones support for Bastion is in public preview as of May. Is this something we want to add to the backlog to add it in the future? Right now, it is pretty limited in what [regions are supported](https://learn.microsoft.com/en-us/azure/reliability/reliability-bastion?toc=%2Fazure%2Fbastion%2FTOC.json#regions-supported) as a heads up.

Apologies, I hadn't spotted it was in preview before logging this! - This came back off a handover with a customer and noticing resources flagged as not being zone resilient.

**Comment by oZakari** [comment]
No worries, @integyjc, it's a good call out to make sure it's on our radar.

**Comment by sebassem** [comment]
> Hey [@sebassem](https://github.com/sebassem), as availability zones support for Bastion is in public preview as of May. Is this something we want to add to the backlog to add it in the future? Right now, it is pretty limited in what [regions are supported](https://learn.microsoft.com/en-us/azure/reliability/reliability-bastion?toc=%2Fazure%2Fbastion%2FTOC.json#regions-supported) as a heads up.

Yes indeed, I added it to our backlog for the Bastion refresh. Thanks @oZakari

**Comment by oZakari** [comment]
Moving this to the upstream Enterprise-Scale repository for tracking.
